src/herder/HerderSCPDriver.cpp

```diff
@@ -187,13 +218,43 @@ HerderSCPDriver::validateValueHelper(uint64_t slotIndex,
 
     bool compat = isSlotCompatibleWithCurrentState(slotIndex);
 
+    auto const& lcl = mLedgerManager.getLastClosedLedgerHeader().header;
+
+    // when checking close time, start with what we have locally
+    lastCloseTime = lcl.scpValue.closeTime;
+
     if (compat)
     {
-        lastCloseTime = mLedgerManager.getLastClosedLedgerHeader()
-                            .header.scpValue.closeTime;
+        if (!checkCloseTime(slotIndex, lastCloseTime, b))
+        {
+            return SCPDriver::kInvalidValue;
+        }
     }
     else
     {
+        if (slotIndex == lcl.ledgerSeq)
+        {
+            // previous ledger
+            if (b.closeTime != lastCloseTime)
+            {
+                return SCPDriver::kInvalidValue;
+            }
+        }
+        else if (slotIndex < lcl.ledgerSeq)
+        {
+            // basic sanity check on older value
+            if (b.closeTime >= lastCloseTime)
+            {
+                return SCPDriver::kInvalidValue;
+            }
+        }
+        else if (!checkCloseTime(slotIndex, lastCloseTime, b))
+        {
+            // new messages must meet the time criteria from lcl
+            // as lcl's close time is a lower bound
```

Comment:
This condition is triggered more generally than what is mentioned in the comment, since `checkCloseTime` also returns false if the time is too far in the future. This should be harmonized: if the comment is right then don't call `checkCloseTime` and just check the lower bound, if the comment is wrong then it should be corrected. In the previous implementation, if we weren't tracking then it would never check if the time is too far in the future.

Reply:
agree, changed the comment to not be misleading